apps/server/src/modules/task/uc/task.uc.ts

```diff
@@ -1,15 +1,43 @@
 import { Injectable } from '@nestjs/common';
-import { PaginationQueryDto } from '../../../shared/core/controller/dto/pagination.query.dto';
-import { Paginated } from '../../../shared/core/types/paginated';
-import { EntityId, ITaskMetadata } from '../entity';
+import { EntityId, IPagination } from '@shared/domain';
+import { Counted } from '@shared/domain/types';
 import { TaskRepo } from '../repo/task.repo';
+import { Task } from '../entity';
+import { SubmissionRepo } from '../repo/submission.repo';
+
+// TODO move to different file
+export type TaskSubmissionsMetaData = { submitted: number; maxSubmissions: number; graded: number };
 
 // filter tasks older than 3 weeks
 @Injectable()
 export class TaskUC {
-	constructor(private taskRepo: TaskRepo) {}
+	constructor(private taskRepo: TaskRepo, private submissionRepo: SubmissionRepo) {}
+
+	async findAllOpenForUser(userId: EntityId, pagination: IPagination): Promise<Counted<Task[]>> {
+		// TODO authorization (user conditions -> permissions?)
+		// TODO get permitted tasks...
+		// TODO have BL from repo here
+
+		const [tasks, total] = await this.taskRepo.findAllOpenByStudent(userId, pagination);
+		return [tasks, total];
+	}
+
+	// TODO move somewhere more private
+	async getTaskSubmissionMetadata(task: Task): Promise<TaskSubmissionsMetaData> {
```

Comment:
for uc we can outsource it by different exports or different export that has the name export const private = { function1, ..}